### PR TITLE
Add gemmini Loop WS kernel support through ACCFG

### PR DESCRIPTION
**Pull request by JosseVanDelm**
This PR implements the following:
* A conversion from linalg generic to Gemmini's weight-stationary kernel
* A "temporary fix" to allow deduplicated RoCC code to be converted to CSRs.
Right now there is no way to create pairs if one of the pairs is deduplicated. 
The `create_pairs` function will try to catch the previous state to create a pair again, but this is not possible if only half of the pair is set in said previous state.
As a first fix, this test assumes all settings are set in pairs, and a constant default is appended to the very first setup op in the state chain (the one without an in_state). This makes sure that a default value is set to rs1 if rs2 is set - but not rs1 - and vice versa.  A more thorough fix for this actually might reset all values at first? Any suggestions welcome.
- [x] add test for this ^^

### CODE REVIEW

**Comment by JosseVanDelm** [comment]
Hi! I'm opening this for review.
I understand the current version of the container does not support the current transform Dialect statements as per @JonasCrols' comments.

**Comment by JosseVanDelm** [comment]
> (I can also hack a sed fix in there if required. 

This is what I went with, since it works pretty well, also over at https://github.com/KULeuven-MICAS/gemmini-rocc-tests/blob/dev/bareMetalMLIR

File: kernels/gemmini/Makefile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+%.ll.mlir: %.postproc.mlir
+	${MLIROPT} --test-lower-to-llvm -o $@ $<
+
+%.ll: %.postproc.mlir
+	${MLIRTRANSLATE} --mlir-to-llvmir -o $@ $<
+
+%.x: %.ll
+	${CC} -x ir $< -c -O3 -o $@ -Wno-override-module --target=riscv64-unknown-elf -mcpu=generic-rv64 -march=rv64gc
+
+allrun: all
```
Comment (JosseVanDelm) [inline kernels/gemmini/Makefile:33]: Note that it does not actually run this, otherwise we'd have to install spike with gemmini extensions

**Review by jorendumoulin — APPROVED** [review]
Okay!

**Pull request merged**